cub/test/catch2_test_device_merge.cu

```diff
@@ -0,0 +1,356 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "insert_nested_NVTX_range_guard.h"
+// above header needs to be included first
+
+#include <cub/device/device_merge.cuh>
+
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/sort.h>
+
+#include <algorithm>
+
+#include "catch2_test_helper.h"
+#include "catch2_test_launch_helper.h"
+#include <test_util.h>
+
+// %PARAM% TEST_LAUNCH lid 0:1:2
+
+DECLARE_LAUNCH_WRAPPER(cub::DeviceMerge::MergePairs, merge_pairs);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceMerge::MergeKeys, merge_keys);
+
+using types = c2h::type_list<std::uint8_t, std::int16_t, std::uint32_t, double>;
+
+// TODO(bgruber): are we interested in 16-bit offset types?
+using offset_types =
+  c2h::type_list</*std::int16_t, std::uint16_t,*/ std::int32_t, std::uint32_t, std::int64_t, std::uint64_t>;
```

Comment:
If we allow smaller than 32-bit offset types, I may need to add extra casts in the kernels to work against integral promotion.

Reply:
Discussion continues below.